optuna/importance/_fanova/_tree.py

```diff
@@ -120,19 +121,24 @@ def _get_marginalized_statistics(self, feature_vector: numpy.ndarray) -> Tuple[f
                     continue
 
                 # If subtree starting from node splits on an active feature, push both child nodes.
-                if (active_features & self._subtree_active_features[node_index]).any():
+                if any(self._subtree_active_features[node_index][active_features].tolist()):
                     for child_node_index in self._get_node_children(node_index):
                         active_nodes.append(child_node_index)
                         active_search_spaces.append(search_spaces)
                     continue
 
             # If node is a leaf or the subtree does not split on any of the active features.
             node_indices.append(node_index)
-            active_features_cardinalities.append(_get_cardinality(search_spaces))
+            active_leaf_search_spaces.append(search_spaces)
 
         statistics = self._statistics[node_indices]
         values = statistics[:, 0]
         weights = statistics[:, 1]
+        active_features_cardinalities = numpy.asarray(active_leaf_search_spaces)
+        active_features_cardinalities = (
+            active_features_cardinalities[:, :, 1] - active_features_cardinalities[:, :, 0]
+        )
+        active_features_cardinalities = numpy.prod(active_features_cardinalities, axis=1)
```

Comment:
We could introduce a `_get_cardinality_batched`, or generalize `_get_cardinality` to handle these cases. Very optional. (And thanks a lot for the PR!)